lib/modules/collapseWhitespace.es6

```diff
@@ -19,42 +19,44 @@ const noTrimWhitespacesInsideElements = new Set([
     'a', 'abbr', 'acronym', 'b', 'big', 'del', 'em', 'font', 'i', 'ins', 'kbd', 'mark', 'nobr', 'rp', 's', 'samp', 'small', 'span', 'strike', 'strong', 'sub', 'sup', 'time', 'tt', 'u', 'var'
 ]);
 
-const whitespacePattern = /\s+/g;
+const startsWithWhitespacePattern = /^\s/;
+const endsWithWhitespacePattern = /\s$/;
+const multipleWhitespacePattern = /\s+/g;
 const NONE = '';
 const SINGLE_SPACE = ' ';
 const validOptions = ['all', 'aggressive', 'conservative'];
 
 /** Collapses redundant whitespaces */
-export default function collapseWhitespace(tree, options, collapseType, tag) {
+export default function collapseWhitespace(tree, options, collapseType, parent) {
     collapseType = validOptions.includes(collapseType) ? collapseType : 'conservative';
 
     tree.forEach((node, index) => {
-        if (typeof node === 'string') {
-            const prevNode = tree[index - 1];
-            const nextNode = tree[index + 1];
-            const prevNodeTag = prevNode && prevNode.tag;
-            const nextNodeTag = nextNode && nextNode.tag;
+        const prevNode = tree[index - 1];
+        const nextNode = tree[index + 1];
 
-            const isTopLevel = !tag || tag === 'html' || tag === 'head';
+        if (typeof node === 'string') {
+            const parentNodeTag = parent && parent.node && parent.node.tag;
+            const isTopLevel = !parentNodeTag || parentNodeTag === 'html' || parentNodeTag === 'head';
             const shouldTrim = (
                 collapseType === 'all' ||
                 isTopLevel ||
                 /*
                  * When collapseType is set to 'aggressive', and the tag is not inside 'noTrimWhitespacesInsideElements'.
                  * the first & last space inside the tag will be trimmed
                  */
-                (
-                    collapseType === 'aggressive' &&
-                    !noTrimWhitespacesInsideElements.has(tag)
-                )
+                collapseType === 'aggressive'
```

Comment:
Here I remove `noTrimWhitespacesInsideElements` determination to the inside of `collapseRedundantWhitespaces` method, in order to handle the edge case mentioned in #145.